service/controller/key/key.go

```diff
@@ -152,6 +153,10 @@ func BlobName(customObject LabelsGetter, role string) string {
 	return fmt.Sprintf("%s-%s-%s", OperatorVersion(customObject), cloudConfigVersion, role)
 }
 
+func BootstrapBlobName(customObject expcapzv1alpha3.AzureMachinePool) string {
```

Comment:
@tuommaki I added a new function to generate the blob name. I realized I was using the old `BlobName` function that would return the same name for all node pools. In this one I'm including cluster name, node pool name and operator version.